js/aspnet.js

```diff
@@ -140,7 +140,8 @@
             id = id || ("dx-" + new Guid());
 
             var render = function(_, container) {
-                var $component = $("#" + id, container)[name](options);
+                var selector = "#" + id.replace(/[^\w-]/g, "\\$&"),
```

Comment:
See also:
https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Regular_Expressions
https://github.com/jquery/jquery/blob/2d4f53416e5f74fa98e0c1d66b6f3c285a12f0ce/src/data.js#L54
https://stackoverflow.com/questions/4688730/what-does-in-str-replacespecials-mean